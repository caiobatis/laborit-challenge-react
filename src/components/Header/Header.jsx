import React from 'react';

export default function Header() {
  return (<>
    <header>
      <div className="container">
        <div className="brand">
          <h1 className="h1">
            Laborit
            <sub>Challenge</sub>
          </h1>
        </div>
      </div>
    </header>
  </>)
}
