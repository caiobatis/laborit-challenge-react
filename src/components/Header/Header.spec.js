import Header from './Header'

describe('Header Component', () => {
  let component
  beforeEach(() => {
    const props = { title: 'lab' }
    component = shallow(<Header {...props} />)
  })
  
  it('should render correctly', () => {
    expect(component).toMatchSnapshot()
    component.unmount()
  })

  it('should render new title', () => {
    let title = 'new title'
    component.setProps({ title })

    expect(component.find('h1').props().children[0]).toEqual(title)
  })
})