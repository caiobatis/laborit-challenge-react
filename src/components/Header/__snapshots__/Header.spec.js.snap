// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="lab"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <header>
        <div
          className="container"
        >
          <div
            className="brand"
          >
            <h1
              className="h1"
            >
              lab
              <sub>
                Challenge
              </sub>
            </h1>
          </div>
        </div>
      </header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="brand"
          >
            <h1
              className="h1"
            >
              lab
              <sub>
                Challenge
              </sub>
            </h1>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="brand"
          >
            <h1
              className="h1"
            >
              lab
              <sub>
                Challenge
              </sub>
            </h1>
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="h1"
            >
              lab
              <sub>
                Challenge
              </sub>
            </h1>,
            "className": "brand",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "lab",
                <sub>
                  Challenge
                </sub>,
              ],
              "className": "h1",
            },
            "ref": null,
            "rendered": Array [
              "lab",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Challenge",
                },
                "ref": null,
                "rendered": "Challenge",
                "type": "sub",
              },
            ],
            "type": "h1",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "header",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <header>
          <div
            className="container"
          >
            <div
              className="brand"
            >
              <h1
                className="h1"
              >
                lab
                <sub>
                  Challenge
                </sub>
              </h1>
            </div>
          </div>
        </header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="brand"
            >
              <h1
                className="h1"
              >
                lab
                <sub>
                  Challenge
                </sub>
              </h1>
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="brand"
            >
              <h1
                className="h1"
              >
                lab
                <sub>
                  Challenge
                </sub>
              </h1>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="h1"
              >
                lab
                <sub>
                  Challenge
                </sub>
              </h1>,
              "className": "brand",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "lab",
                  <sub>
                    Challenge
                  </sub>,
                ],
                "className": "h1",
              },
              "ref": null,
              "rendered": Array [
                "lab",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Challenge",
                  },
                  "ref": null,
                  "rendered": "Challenge",
                  "type": "sub",
                },
              ],
              "type": "h1",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "header",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
